trillian/migrillian/core/controller.go

```diff
@@ -242,21 +246,28 @@ func (c *Controller) verifyConsistency(ctx context.Context, root *types.LogRootV
 	return nil
 }
 
-// runSubmitter obtaines CT log entry batches from the controller's channel and
-// submits them through Trillian client. Returns when the channel is closed.
-func (c *Controller) runSubmitter(ctx context.Context) {
+// runSubmitter obtains CT log entry batches from the controller's channel and
+// submits them through Trillian client. Returns when the channel is closed, or
+// the client returns a non-recoverable error (an example of a recoverable
+// error is when Trillian write quota is exceeded).
+func (c *Controller) runSubmitter(ctx context.Context) error {
 	treeID := c.plClient.tree.TreeId
 	for b := range c.batches {
 		entries := float64(len(b.Entries))
 		metrics.entriesSeen.Add(entries, c.label)
 
 		end := b.Start + int64(len(b.Entries))
-		// TODO(pavelkalinnikov): Retry with backoff on errors.
 		if err := c.plClient.addSequencedLeaves(ctx, &b); err != nil {
-			glog.Errorf("%d: failed to add batch [%d, %d): %v", treeID, b.Start, end, err)
+			// addSequencedLeaves failed to submit entries despite retries. At this
+			// point there is not much we can do. Seemingly the best strategy is to
+			// shut down the Controller.
+			// TODO(pavelkalinnikov): Restart Controller and/or expose some metrics
+			// allowing a log operator to set up alerts and react accordingly.
+			return fmt.Errorf("failed to add batch [%d, %d): %v", b.Start, end, err)
 		} else {
```

Comment:
No need for the `else{}` block now that the `if {}` block returns

Reply:
Done.